### PR TITLE
fix: statically register formatting and semantic tokens capabilities

### PR DESCRIPTION
**Pull request by MrLego8-9**
A fix for #799.
Instead of dynamically registering formatting and semantic tokens capabilities, register them statically.

The toggles in the config works the same, for formatting and semantic highlighting. This, however, improves compatibility with some editors that do not support dynamic registering in their LSP client implementation.

I have tested the changes in VSCode (CodeOSS), Helix and Kate (does not support dynamic registration).

### CODE REVIEW

**Comment by Myriad-Dreamin** [comment]
Could you inline these two variable? The comment above is not needed as well.

File: crates/tinymist/src/init.rs [hunk 1]
```diff
@@ -150,22 +150,15 @@ impl Initializer for SuperInit {
...
-        };
-        let document_formatting_provider = match service.config.formatter_mode {
-            FormatterMode::Typstyle | FormatterMode::Typstfmt
-                if !const_config.doc_fmt_dynamic_registration =>
-            {
-                Some(OneOf::Left(true))
-            }
-            _ => None,
-        };
+        let semantic_tokens_provider = Some(
```
Comment (Myriad-Dreamin) [inline crates/tinymist/src/init.rs:153]: Could you inline these two variable? The comment above is not needed as well.

Comment (MrLego8-9) [reply crates/tinymist/src/init.rs]: Fixed

**Comment by Myriad-Dreamin** [comment]
I have tested it on my vscode, it doesn't not work if we always register them statically, so I turn to register them only if dynamic registering is not supported. Since we have made a mistake, I might delay to merge it in v0.12.2+ to reduce risk to break things.

**Comment by MrLego8-9** [comment]
> I have tested it on my vscode, it doesn't not work if we always register them statically, so I turn to register them only if dynamic registering is not supported. Since we have made a mistake, I might delay to merge it in v0.12.2+ to reduce risk to break things.

Sure, no problem, if you think it's better to wait in order not to break things, go ahead.

**Pull request merged**